wasi/2023/WASI-11-02.md

```diff
@@ -30,4 +30,199 @@ The meeting will be on a zoom.us video conference.
+
+**Pat Hickey**: In WebAssembly org, we think we are very close to settling those WIT specs. We have a project board that keeps this project tracking board that we use to keep track of various Preview 2 things between individual contributors. wasi-sockets tweaks and wasi-http implementations needed to be reflected up to WebAssembly org. 
+
+Whatever changes we have to do between wasmtime or cap-std we will do to sync with the WebAssembly org.
+
+Are there any other comments before we move on to the next presentation?
+
+###Jco implementation status update (Guy Bedford, 20 mins)
+
+**Guy Bedford**: JCO implementation of WASI. Update on where we are currently and how that implementation is working. For those of you who don’t know, JCO is a JavaScript toolchain for Wasm components with npm install -g jco
+
+**Guy Bedford**: What we have been adding is full WASI support for these components.
+
+**Guy Bedford**: Who is working on it? Myself, I have been at Fastly for a year and I have a background as a Node.js Collaborator and I am a TC-39 delegate. As well as help from Microsoft engineers Yoshua Wuyts, and Wassim Chegam, who is also a Node.js Collaborator.
+
+**Guy Bedford**: WASI on Node.js. Our hope is that the experience we get when implementing WASI with jco is that we can have WASI work well in the Node.js project. I am also a maintainer of the Node.js project.
+
+**Guy Bedford**: JCO recently added support for running command line components with the CLI world and now we are working on adding a serve command similar to wasmtime. This will let us demonstrate the wasi-http-proxy world.
+
+**Guy Bedford**: Our implementation goals will let us get full support for all component model features. Now the focus is on full WASI conformance for both the serverside in Node.js and on the browser side. We are pulling in the wasm test suite and making sure we can pass all of those tests.
+
+**Guy Bedford**: Experimental support right now while figuring out the architecture. Partial implementations on I/O. HTTP supports basic request/response. HTTP on the web is not yet supported. Sockets is not yet supported on server-side but should have this updated soon. Clocks shouldn’t be too tricky to finish up. Filesystem has partial support and internally JCO uses it’s own filesystem implementation.
+
+**Guy Bedford**: The current implementation is experimental but come to good grips with the architecture at this point. Expecting to reach full support for WASI APIs on Node.js /JS server runtimes by EOY. Delay into Jan/Feb may be possible.
+
+**Syrus Akbary**: Is JCO for supporting the core node libraries?
+
+**Guy Bedford**: When you say using the Node.js API do you mean building on top of the Node libraries?
+
+**Syrus Akbary**:  <notetaker missed this>
```

Comment:
@syrusakbary The notetaker missed this, if possible can you fill it in?

Reply:
I do not remember tbh, let me make an educated guess:

```suggestion
**Syrus Akbary**:  Not exactly, I meant being able to run Node.js Applications in JCO
```

Note: as I commented previously, I strongly believe it will be worth taking automated notes in the meetings instead of having a typewriter that could miss it. I think that would easily solve problems such as this one